### ISSUE TITLE
Timing issue at start with AbstractEventSource

### ISSUE THREAD

**Issue by shawkins** [java-operator-sdk/java-operator-sdk#417]
This is a minor issue.  Event sources can start receiving events before the controller init is called.  This requires a null check of the eventHandler or you'll get npe exceptions for a while.  Perhaps AbstractEventSource could implement EventHandler and take care of the relevant null checks.

Related to this should the AbstractEventSource.eventHandler field be volatile?

**Comment by metacosm** [comment]
Is this still going on in the latest version?

**Comment by shawkins** [comment]
I haven't checked.  The event source is an application scoped bean.  Events can start flowing as a side effect of a quarkus onStart hook.  Should the operator init be guaranteed to happen before that?  And/or do you want this as a quarkus specific issue?

**Comment by lburgazzoli** [comment]
I think now you should override [EventSource.start](https://github.com/java-operator-sdk/java-operator-sdk/blob/master/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/EventSource.java#L11) and [EventSource.close](https://github.com/java-operator-sdk/java-operator-sdk/blob/master/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/EventSource.java#L18) for a proper lifecycle

**Comment by metacosm** [comment]
Can we close this issue?

**Comment by shawkins** [comment]
> Can we close this issue?

Yes.  Based upon our current code, we'll just stick with a null check.  We don't want to process any events until all of the informers have started/synced.

**Comment by metacosm** [comment]
I'll add a null check to `AbstractEventSource` regardless. Thank you!

**Comment by shawkins** [comment]
> I'll add a null check to AbstractEventSource regardless. Thank you!

Do you mean have AbstractEventSource implement EventHandler and handle the null check in AbstractEventSource.handleEvent?
